x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -701,7 +701,12 @@ protected PhysicalPlan rule(OrderExec plan) {
                 EsQueryExec exec = (EsQueryExec) plan.child();
                 QueryContainer qContainer = exec.queryContainer();
 
-                for (Order order : plan.order()) {
+                // Reverse traversal together with the upwards fold direction ensures that sort clauses are added in reverse order of
+                // precedence. E.g. for the plan `OrderBy[a desc,b](OrderBy[a asc,c](EsExec[...]))`, `prependSort` is called with the
+                // following sequence of arguments: `c`, `a asc`, `b`, `a desc`. The resulting sort order is `a desc,b,c`.
+                for (int i = plan.order().size() - 1; i >= 0; i--) {
+                    Order order = plan.order().get(i);
```

Comment:
Imo, it's cleaner to use the `listIterator` instead of the index
```
for (ListIterator<Order> i = plan.order().listIterator(); i.hasPrevious(); ) {
   Order order = i.previous();
```

Reply:
👍 I was looking for something to avoid the index lookup but was not aware of `ListIterator`.